tests/push/test_productid_updates_versions.py

```diff
@@ -79,22 +68,22 @@ def test_updates_product_versions(monkeypatch, tmpdir):
     ctrl.insert_repository(repo7)
     ctrl.insert_repository(repo8)
 
-    # Set up cert parser to use a fake cert object (saves us having to explicitly
-    # generate certs with certain values)
-    monkeypatch.setattr(
-        rhsm.certificate, "create_from_file", lambda _: FakeCert(["a", "d"])
-    )
-
     # make a fake productid.
-    # content doesn't matter since we patched the cert parser, it just has
+    # content doesn't matter since we mock the ProductIDs, it just has
```

Comment:
Err... sorry to be pedantic again about this comment (but you have said in other reviews that you like it).

I don't think "mock" is the right word here, and I think using it contributes to confusion around what mocking is.  Well, I don't think it's a very well-defined term, but I agree with Wikipedia's leading sentence: "A mock object is an object that imitates a production object in limited ways."  You are providing *real* ProductId objects here, which are going to work in exactly the same way as ProductId objects loaded out of a cert.  You're using the API of pushsource in a normal and valid way and I don't see how it should be considered "mocking".  You're just providing your own data which skips the default behavior of trying to load that data from a file.

If I may give an analogy, if there was a function with a signature like:

```python
def load_config(path: str = "/etc/some-default.cfg"):
    ...
```

...and some code were to call that function like `cfg = load_config("my-non-default.cfg")`, I don't think that should be called "mocking" the path or the config?

I believe it's better for the comment to say "we provide our own ProductIds", and also for us to think about the code in this way.  Or if you wanted to emphasize that this is overriding the default method of obtaining the objects, you could even use the term "inject" per [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection), "a programming technique in which an object or function receives other objects or functions that it requires, as opposed to creating them internally".